src/documents/tests/test_consumer.py

```diff
@@ -1,11 +1,24 @@
 from django.test import TestCase
 from unittest import mock
 from tempfile import TemporaryDirectory
+from io import StringIO
+import pytest
 
 from ..consumer import Consumer
 from ..models import FileInfo, Tag
 
 
+def error_on_call_n(error, n):
+    """Returns an object which raises an error on the n-th call"""
+    count = [0]
```

Comment:
I should have really put a comment there. Python scoping rules are really weird. The code without the list fails:

def error_on_call_n(error, n):
    count = 0
    def error_object(*args, **kwargs):
        if count == n:
            raise error
        else:
            count += 1
    return error_object

with an UnboundLocalError: local variable 'count' referenced before assignment. Because of the assignment `count += 1` python understands `count` as a local variable (with the scope of the function `error_object`), but when checking the if condition `count == n` the local variable `count` is not existent. Changing count to a list changes count[0] += 1 from an "variable assignment" to an "operation on an object" so python does not understand count as a local variable but ("correctly") looks it up (so that it finds it in the scope of `error_on_call_n`).

The "correct" solution is apparently to tell python that count is nonlocal:

def error_on_call_n(error, n):
    count = 0
    def error_object(*args, **kwargs):
        nonlocal count
        if count == n:
            raise error
        else:
            count += 1
    return error_object

but to be honest I have never seen a `nonlocal` in the wild. While the list solution is confusing why exactly it was chosen, I think every python developer understands what is happening. With `nonlocal` I would assume that more people would have to lookup what that even is.

Should I change that to the `nonlocal` solution?